files/scponly-chroot.sh

```diff
@@ -0,0 +1,31 @@
+#! /bin/bash
+
+ALTROOT="$1"; shift
+BINARIES="$@"
+
+for b in $BINARIES
+do
+    d=$(dirname "${b}")
+    if [ ! -d "${ALTROOT}${d}" ]; then
+        /bin/mkdir -p "${ALTROOT}${d}"
+    fi
+    /bin/cp "${b}" "${ALTROOT}${d}"
+
+    LIBS=$(ldd "${b}" | awk '{ print $3 }' | egrep -v -v ^'\(')
+
+    for i in ${LIBS}; do
+        d=$(dirname "${i}")
+        if [ ! -d "${ALTROOT}${d}" ]; then
+            /bin/mkdir -p "${ALTROOT}${d}"
+        fi
+            /bin/cp "${i}" "${ALTROOT}${d}"
+    done
+
+    ld=$(ldd "${b}" | grep 'ld-linux' | awk '{ print $1 }')
+    d=$(dirname "${ld}")
+
+    if [ ! -d "${ALTROOT}${d}" ]; then
+        /bin/mkdir -p "${ALTROOT}${d}"
+    fi
+    /bin/cp "${ld}" "${ALTROOT}${d}"
+done
```

Comment:
You need to add the following as well to get the chroot to work properly:
```bash
# Misc fixes
/bin/cp /lib64/libnss_* ${ALTROOT}/lib64/

# Devices
mkdir -p ${ALTROOT}/dev
mknod -m 666 ${ALTROOT}/dev/null c 1 3
mknod -m 666 ${ALTROOT}/dev/zero c 1 5
mknod -m 666 ${ALTROOT}/dev/tty c 5 0
mknod -m 444 ${ALTROOT}/dev/random c 1 8
mknod -m 444 ${ALTROOT}/dev/urandom c 1 9
```

Reply:
For my own knowledge, how did you figure out to do this? I never found anything like this in the documentation.

Reply:
A little of trial and error. For the misc fixes I figured this out by trying a few this manually in the chroot. For the nodes, I found a reference online somewhere and I recalled that chroots sometimes need these device nodes for it to work properly.

Reply:
does ldd not output lib64 libraries that are required? What are the libnss_* libraries used by?